src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/AndroidNetworkInterface.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Net;
+
+namespace System.Net.NetworkInformation
+{
+    /// <summary>
+    /// Implements a NetworkInterface on Android.
+    /// </summary>
+    internal sealed class AndroidNetworkInterface : UnixNetworkInterface
+    {
+        private OperationalStatus _operationalStatus;
+        private bool _supportsMulticast;
+        private long _speed;
+        internal int _mtu;
+        private NetworkInterfaceType _interfaceType = NetworkInterfaceType.Unknown;
+        private readonly AndroidIPInterfaceProperties _ipProperties;
+
+        internal AndroidNetworkInterface(string name, int index) : base(name)
+        {
+            _index = index;
+            _ipProperties = new AndroidIPInterfaceProperties(this);
+        }
+
+        public static unsafe NetworkInterface[] GetAndroidNetworkInterfaces()
```

Comment:
Wouldn't be better to have this implemented in `NetworkInterfacePal` class and construct read-only AndroidNetworkInterface from there?

Reply:
I just modified the existing `LinuxNetworkInterface` implementation. I can definitely move the code into the `Pal` class. It will definitely be cleaner 👍